modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/display/context/ViewConflictsDisplayContext.java

```diff
@@ -245,19 +247,68 @@ public String getRedirect() {
 			if (!conflictInfo.isResolved()) {
 				JSONArray actionsJSONArray = JSONFactoryUtil.createJSONArray();
 
-				if (_ctCollection.getCtCollectionId() ==
-						_activeCtCollectionId) {
+				String editURL = _ctDisplayRendererRegistry.getEditURL(
+					_httpServletRequest, ctEntry);
 
-					String editURL = _ctDisplayRendererRegistry.getEditURL(
-						_httpServletRequest, ctEntry);
+				if (Validator.isNotNull(editURL)) {
+					String editInPublicationURL = editURL;
+
+					if (_activeCtCollectionId !=
+							_ctCollection.getCtCollectionId()) {
+
+						editInPublicationURL =
+							PublicationsPortletURLUtil.getHref(
+								_renderResponse.createActionURL(),
+								ActionRequest.ACTION_NAME,
+								"/change_tracking/checkout_ct_collection",
+								"redirect", editURL, "ctCollectionId",
+								String.valueOf(
+									_ctCollection.getCtCollectionId()));
+					}
+
+					actionsJSONArray.put(
+						JSONUtil.put(
+							"href", editInPublicationURL
+						).put(
+							"label",
+							_language.format(
+								_httpServletRequest, "edit-in-x",
+								new Object[] {_ctCollection.getName()}, false)
+						).put(
+							"symbol", "pencil"
+						));
+
+					T productionModel = _ctDisplayRendererRegistry.fetchCTModel(
+						modelClassNameId, conflictInfo.getTargetPrimaryKey());
+
+					if (productionModel != null) {
+						String editInProductionURL =
+							_ctDisplayRendererRegistry.getEditURL(
+								_httpServletRequest, productionModel,
+								modelClassNameId);
+
+						if (_activeCtCollectionId !=
+								CTConstants.CT_COLLECTION_ID_PRODUCTION) {
+
+							editInProductionURL =
+								PublicationsPortletURLUtil.getHref(
+									_renderResponse.createActionURL(),
+									ActionRequest.ACTION_NAME,
+									"/change_tracking/checkout_ct_collection",
+									"redirect", editInProductionURL,
+									"ctCollectionId",
+									String.valueOf(
+										CTConstants.
+											CT_COLLECTION_ID_PRODUCTION));
+						}
 
-					if (Validator.isNotNull(editURL)) {
 						actionsJSONArray.put(
 							JSONUtil.put(
-								"href", editURL
+								"href", editInProductionURL
```

Comment:
@stsquared99 This part of logic will overwrite whatever you put in earlier which means the values calculated earlier useless,  then we should avoid the first part calculation if you are going to be in this block, this way, we can have better performance.

Tina.

Reply:
@tinatian Sorry, I don't follow what you mean here. Do you mean the overwrite at L293? Because in that case, the earlier calculated value is not being discarded, it is being set as a parameter on the new url.

Reply:
@stsquared99 Both https://github.com/liferay-core-infra/liferay-portal/blob/6580236ed02d54f1a91cb89d4051b1597ee9ed18/modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/display/context/ViewConflictsDisplayContext.java#L307 and https://github.com/liferay-core-infra/liferay-portal/blob/6580236ed02d54f1a91cb89d4051b1597ee9ed18/modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/display/context/ViewConflictsDisplayContext.java#L271 are setting a property with same name "herf" to json. The second one is overwriting the first one.

Reply:
@tinatian Ah. These are separate json objects being added to an array. So the second one is an additional item in the array, not overwriting the first one.

Reply:
@stsquared99 I see... then could you please move the following logic into a method? This way we won't see hard coded strings like "href" more than time?
**JSONUtil.put(
								"href", editInProductionURL
							).put(
								"label",
								_language.get(
									_httpServletRequest, "edit-in-production")
							).put(
								"symbol", "pencil"
							)**